src/Common.cpp

```diff
@@ -166,10 +167,49 @@ std::vector<std::string> NAS2D::split(std::string str, char delim /*= ','*/, boo
 	ss.clear();
 
 	std::string curString{};
-    while(std::getline(ss, curString, delim)) {
-        if(skip_empty && curString.empty()) { continue; }
+	while (std::getline(ss, curString, delim))
+	{
+		if (skip_empty && curString.empty()) { continue; }
 		result.push_back(curString);
-    }
+	}
+	result.shrink_to_fit();
+	return result;
+}
+
+
+std::string NAS2D::join(std::vector<std::string> strs, char delim, bool skip_empty /*= true*/)
+{
+	auto acc_op = [](const std::size_t& a, const std::string& b) -> std::size_t { return a + static_cast<std::size_t>(1u) + b.size(); };
+	auto total_size = std::accumulate(std::begin(strs), std::end(strs), static_cast<std::size_t>(0u), acc_op);
+	std::string result;
+	result.reserve(total_size);
+
+	for (auto iter = std::begin(strs); iter != std::end(strs); ++iter)
+	{
+		if (skip_empty && (*iter).empty()) { continue; }
+		result += (*iter);
+		if (iter != std::end(strs) - 1)
```

Comment:
Neither `std::vector::count()` nor `std::string::count` is a thing. I think you mean `size`. Even then, calling `size` over `empty` is less efficient when checking against zero.

Also see above for negative conditions. In addition, short-circuiting on a bool is more efficient than a sequence of dereferencing an iterator, then a function call, then negating the result.

Reply:
Err, yes, I meant `size()`.

The use of `size()` doesn't replace `empty()`. Both are used. The difference is the delimiter is no longer conditional within the loop. The loop iteration is beginning tested, by peeling the first iteration off the loop, rather than end tested within the loop. I didn't point it out, but there is a `+ 1` added to the `std::begin(strs)` loop start condition.

You have a point about swapping the order of the `bool` and the call to `empty()`.

I also just realized my suggestion doesn't quite handle the case of skipping over blank strings without a delimiter in the join when the first string is empty. For that, you may need to initialize to the first non-empty element and start the loop past the first non-empty element.

I suppose if you had an iterator that skipped over empty strings, you could template the method to work with any iterable, rather than hardcode for vectors. That would also eliminate the need for the boolean parameter. The design starts being a lot more work though.